safehttp/plugins/csp/csp_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csp
+
+import (
+	"context"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/go-safeweb/safehttp"
+	"github.com/google/go-safeweb/safehttp/safehttptest"
+)
+
+type dummyReader struct{}
+
+func (dummyReader) Read(b []byte) (int, error) {
+	for i := range b {
+		b[i] = 41
+	}
+	return len(b), nil
+}
+
+func TestMain(m *testing.M) {
+	randReader = dummyReader{}
+	os.Exit(m.Run())
+}
+
+func TestSerialize(t *testing.T) {
+	tests := []struct {
+		name       string
+		policy     Policy
+		wantString string
+	}{
+		{
+			name:       "StrictCSP",
+			policy:     StrictCSPBuilder{}.Build(),
+			wantString: "object-src 'none'; script-src 'unsafe-inline' https: http: 'nonce-super-secret'; base-uri 'none'",
+		},
+		{
+			name:       "StrictCSP with strict-dynamic",
+			policy:     StrictCSPBuilder{StrictDynamic: true}.Build(),
+			wantString: "object-src 'none'; script-src 'unsafe-inline' https: http: 'nonce-super-secret' 'strict-dynamic'; base-uri 'none'",
+		},
+		{
+			name:       "StrictCSP with unsafe-eval",
+			policy:     StrictCSPBuilder{UnsafeEval: true}.Build(),
+			wantString: "object-src 'none'; script-src 'unsafe-inline' https: http: 'nonce-super-secret' 'unsafe-eval'; base-uri 'none'",
+		},
+		{
+			name:       "StrictCSP with set base-uri",
+			policy:     StrictCSPBuilder{BaseURI: "https://example.com"}.Build(),
+			wantString: "object-src 'none'; script-src 'unsafe-inline' https: http: 'nonce-super-secret'; base-uri https://example.com",
+		},
+		{
+			name:       "StrictCSP with report-uri",
+			policy:     StrictCSPBuilder{ReportURI: "https://example.com/collector"}.Build(),
+			wantString: "object-src 'none'; script-src 'unsafe-inline' https: http: 'nonce-super-secret'; base-uri 'none'; report-uri https://example.com/collector",
+		},
+		{
+			name:       "FramingCSP",
+			policy:     FramingPolicy(false, ""),
```

Comment:
You can see that this is not very readable. When I read `csp.FramingPolicy(false, "")`, I have no idea what these arguments mean. Could you find a way to make it more readable?

Reply:
I decided to make a Builder instead, like the `StrictCSPBuilder`.